recipes/libnl/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+import os, glob
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+class LibNlConan(ConanFile):
+    name = "libnl"
+    description = "A collection of libraries providing APIs to netlink protocol based Linux kernel interfaces."
+    topics = "conan", "netlink"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.infradead.org/~tgr/libnl/"
+    license = "LGPL-2.1-only"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"fPIC": [True, False], "shared": [True, False]}
+    default_options = {"fPIC": True, "shared": False}
+    build_requires = ( "flex/2.6.4", "bison/3.5.3" )
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Libnl is only supported on Linux")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration("Libnl is only supported on Linux")
        if self.options.shared:
            del self.options.fPIC
```

Reply:
Hi Ulian,
I thought the AutoToolsBuildEnvironment handled the fPIC options automatically. If not, then why should we remove fPIC if shared == True? Because fPIC wil be set to True anyway when shared == True ?

Reply:
The point is more about package id. it doesn't make sense using shared + non fPIC, thus offering an useless option which reflects in the package id is not good.